### PR TITLE
Fix scheduling regression

### PR DESCRIPTION
**Pull request by zmc**
I expect these initial two commits to fail CI; I believe #1776 regressed somehow, but haven't fully debugged it yet.

### CODE REVIEW

**Comment by zmc** [comment]
I really want to get the CI fixed, and have made a lot of progress squashing bugs that seem to only pop up in GitHub Actions, but I've got to stop for now. Currently we're failing because Ubuntu can't find `python-apt`, which I've just verified does in fact exist...

**Comment by zmc** [comment]
Finally got a pass. TIL that `docker-compose build` does not update base images, even if they use `:latest`.

File: teuthology/suite/run.py [hunk 1]
```diff
@@ -583,12 +587,12 @@ def schedule_suite(self):
                                seed=self.args.seed)
         generated = len(configs)
         log.info(f'Suite {suite_name} in {suite_path} generated {generated} jobs (not yet filtered or merged)')
-        configs = config_merge(configs,
+        configs = list(config_merge(configs,
```
Comment (batrick) [inline teuthology/suite/run.py:590]: >    The var being left as a generator led to all jobs being filtered out of    some suites, rendering them broken.

! Can you show me an example?

**Review by batrick — COMMENTED** [review]
rest looks okay to me; but I'm not that familiar with some of that code

File: teuthology/suite/run.py (hunk 1 above)
Comment (zmc) [reply teuthology/suite/run.py:590]: Try this:
`teuthology-suite -v --teuthology-branch main --ceph-repo https://github.com/ceph/ceph.git --suite-repo https://github.com/ceph/ceph.git -c main -m testnode -n 100 --suite teuthology:no-ceph --suite-branch main --dry-run`

Comment (batrick) [reply teuthology/suite/run.py:590]: :open_mouth: is there some esoteric python gotcha I tripped on or is this just as baffling to you?

Comment (zmc) [reply teuthology/suite/run.py:590]: I didn't step through everything as thoroughly as I could have so I don't know exactly how the generator is being exhausted. But, generators in python can only be iterated over once:
```
>>> l = [n for n in range(5)]  # just a regular list
>>> list(l)
[0, 1, 2, 3, 4]
>>> list(l)
[0, 1, 2, 3, 4]
>>> g = (n for n in range(5))  # this one is a generator
>>> g
<generator object <genexpr> at 0x10464f7b0>
>>> list(g)
[0, 1, 2, 3, 4]
>>> list(g)
[]  # empty!
```

Comment (batrick) [reply teuthology/suite/run.py:590]: Yes I'm aware of that behavior. I don't have an explanation other than a bug in Python. The `configs` generator is only iterated once AFAICT.

Comment (batrick) [reply teuthology/suite/run.py:590]: Oh, I see it's because of `-n`. There is a while loop around `collect_jobs`.

Guess we can't use a generator!

**Review by batrick — APPROVED** [review]

**Pull request merged**